src/units/abilities.cpp

```diff
@@ -342,7 +342,8 @@ std::vector<std::tuple<std::string, t_string, t_string, t_string>> unit::ability
 
 	for (const config::any_child ab : this->abilities_.all_children_range())
 	{
-		bool active = ability_active(ab.key, ab.cfg, loc);
+		bool loc_is_unshrouded = display::get_singleton()->shrouded(loc) == false;
+		bool active = loc_is_unshrouded && ability_active(ab.key, ab.cfg, loc);
```

Comment:
At the first moment, I thought to add this check `display::get_singleton()->shrouded(loc) == false` within `ability_active`, but since this function is used everywhere, I think it's safer to make this check only here.

Reply:
Thanks for keeping it to the minimum necessary codepaths. This function is only called from the sidebar's report code, so it should be a safe assumption here.